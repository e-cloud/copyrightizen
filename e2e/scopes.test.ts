import { runCopyrightizen } from "./runCopyrghtizen";
import { copyReferenceDirectory } from './utils'
import { compare } from 'dir-compare';

describe('Scopes management tests', () => {

  describe('scopes precedence', () => {
    test('it should handle scopes precedence correctly', async () => {
      const { tmpDir, refDir } = await copyReferenceDirectory('scopes-precedence')
      const result = await runCopyrightizen(tmpDir);

      const compareResult = await compare(refDir, tmpDir, { compareContent: true, noDiffSet: true })

      expect(result.status).toEqual(0)
      expect(compareResult.same).toEqual(true)
      expect(compareResult).toMatchSnapshot()
    })
  })

  describe('multiple scopes intersection', () => {
    test('it should handle scopes correctly', async () => {
      const { tmpDir, refDir } = await copyReferenceDirectory('multi-intersected-scopes')
      const result = await runCopyrightizen(tmpDir);

      const compareResult = await compare(refDir, tmpDir, { compareContent: true, noDiffSet: true })

      expect(result.status).toEqual(0)
      expect(compareResult.same).toEqual(true)
      expect(compareResult).toMatchSnapshot()
    })
  })

})
