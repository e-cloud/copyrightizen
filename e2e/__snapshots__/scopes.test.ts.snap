// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scopes management tests multiple scopes intersection it should handle scopes correctly 1`] = `
Object {
  "differences": 0,
  "differencesDirs": 0,
  "differencesFiles": 0,
  "distinct": 0,
  "distinctDirs": 0,
  "distinctFiles": 0,
  "equal": 12,
  "equalDirs": 4,
  "equalFiles": 8,
  "left": 0,
  "leftDirs": 0,
  "leftFiles": 0,
  "right": 0,
  "rightDirs": 0,
  "rightFiles": 0,
  "same": true,
  "total": 12,
  "totalDirs": 4,
  "totalFiles": 8,
}
`;

exports[`Scopes management tests scopes precedence it should handle scopes precedence correctly 1`] = `
Object {
  "differences": 0,
  "differencesDirs": 0,
  "differencesFiles": 0,
  "distinct": 0,
  "distinctDirs": 0,
  "distinctFiles": 0,
  "equal": 6,
  "equalDirs": 0,
  "equalFiles": 6,
  "left": 0,
  "leftDirs": 0,
  "leftFiles": 0,
  "right": 0,
  "rightDirs": 0,
  "rightFiles": 0,
  "same": true,
  "total": 6,
  "totalDirs": 0,
  "totalFiles": 6,
}
`;
